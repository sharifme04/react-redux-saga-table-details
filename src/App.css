.spin-size {
  width: 7rem;
  height: 7rem;
}
