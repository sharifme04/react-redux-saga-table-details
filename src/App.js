import React, { useEffect } from "react";
import { useDispatch, useSelector } from "react-redux";
import { Switch, Route } from "react-router-dom";
import { templatesRequested } from "./app/actions/templates";
import TemplateOverview from "./app/components/TemplateOverview";
import TemplateDetails from "./app/components/TemplateDetails";
import "./App.css";

function App() {
  const dispatch = useDispatch();
  const templates = useSelector((state) => state.templates);
  useEffect(() => {
    dispatch(templatesRequested());
  }, [dispatch]);

  return (
    <div className="container  mt-2">
      {templates?.error ? (
        <div className="alert alert-danger">
          <strong>Error!Reload again!! </strong>
          {templates?.error?.message}
        </div>
      ) : (
        <Switch>
          <Route
            exact
            path="/"
            render={(props) => <TemplateOverview {...templates} {...props} />}
          />
          <Route
            path="/:templateId"
            render={(props) => <TemplateDetails {...templates} {...props} />}
          />
        </Switch>
      )}
    </div>
  );
}

export default App;
