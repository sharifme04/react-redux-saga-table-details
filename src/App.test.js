import React from "react";
import { MemoryRouter } from "react-router-dom";
import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";
import App from "./App";

const mockStore = configureMockStore();
const store = mockStore({});

const fakeTemplates = {
  isLoading: false,
  templates: {
    message: "get_templates_success",
    status: 0,
    _data: [
      {
        conditions: null,
        name: "basic_user",
        optionalDocuments: {
          assetType: {
            conditions: { topLevelDomain: ".edu" },
            values: { emailAddress: true },
          },
        },
        partnerId: "59c25c00-78cd-11ea-ac9b-bf680aaa6d4b",
        requiredDocuments: {
          assetType: {
            conditions: { topLevelDomain: ".edu" },
            values: { emailAddress: true },
          },
        },
        templateId: "487c5b70-9516-11ea-bbfa-ed3b490e43f0",
        type: "basic",
        validFrom: "04-05-2020",
        validTo: "21-05-2020",
        version: 2,
      },
    ],
  },
};

test("should render initial layout", () => {
  const component = shallow(
    <Provider store={store}>
      <MemoryRouter initialEntries={["/"]}>
        <App />
      </MemoryRouter>
    </Provider>
  );
  expect(component.exists()).toBe(true);
});

test("snapshot renders", () => {
  const component = renderer.create(
    <Provider store={store}>
      <MemoryRouter initialEntries={["/"]}>
        <App />
      </MemoryRouter>
    </Provider>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("snapshot data renders", () => {
  const component = renderer.create(
    <Provider store={store}>
      <MemoryRouter initialEntries={["/"]}>
        <App {...fakeTemplates} />
      </MemoryRouter>
    </Provider>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
