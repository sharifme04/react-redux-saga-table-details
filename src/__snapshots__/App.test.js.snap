// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot data renders 1`] = `
<div
  className="container  mt-2"
>
  <div
    className="container"
  >
    <h2>
      Template Overview
    </h2>
    <hr />
    <div
      className="text-center"
    >
       ohh!!! no template available !!
    </div>
  </div>
</div>
`;

exports[`snapshot renders 1`] = `
<div
  className="container  mt-2"
>
  <div
    className="container"
  >
    <h2>
      Template Overview
    </h2>
    <hr />
    <div
      className="text-center"
    >
       ohh!!! no template available !!
    </div>
  </div>
</div>
`;
